tests/test_modeling_flax_vision_text_dual_encoder.py

```diff
@@ -0,0 +1,382 @@
+        pixel_values = floats_tensor(
+            [
+                batch_size,
+                model.config.vision_config.num_channels,
+                model.config.vision_config.image_size,
+                model.config.vision_config.image_size,
+            ]
+        )
+        input_ids = ids_tensor([batch_size, 4], model.config.text_config.vocab_size)
+        attention_mask = random_attention_mask([batch_size, 4])
+        inputs = {
+            "pixel_values": pixel_values,
+            "input_ids": input_ids,
+            "attention_mask": attention_mask,
+        }
+
+        return model, inputs
+
+    def get_vision_text_model(self, vision_config, text_config):
+        vision_model = FlaxViTModel(vision_config)
+        text_model = FlaxBertModel(text_config)
+        return vision_model, text_model
+
+    def prepare_config_and_inputs(self):
+        vit_model_tester = FlaxViTModelTester(self)
+        bert_model_tester = FlaxBertModelTester(self)
+        vision_config_and_inputs = vit_model_tester.prepare_config_and_inputs()
+        text_config_and_inputs = bert_model_tester.prepare_config_and_inputs()
+
+        vision_config, pixel_values = vision_config_and_inputs
+
+        text_config, input_ids, token_type_ids, attention_mask = text_config_and_inputs
+
+        # make sure that cross attention layers are added
+        return {
+            "text_config": text_config,
+            "vision_config": vision_config,
+            "pixel_values": pixel_values,
+            "attention_mask": attention_mask,
+            "text_config": text_config,
+            "input_ids": input_ids,
+            "token_type_ids": token_type_ids,
+        }
+
+
+@require_torch
+class FlaxCLIPVisionBertModelTest(VisionTextDualEncoderMixin, unittest.TestCase):
+    def get_pretrained_model_and_inputs(self):
+        model = FlaxVisionTextDualEncoderModel.from_vision_text_pretrained(
+            "hf-internal-testing/tiny-random-clip",
+            "hf-internal-testing/tiny-bert",
+            vision_from_pt=True,
+            text_from_pt=True,
+        )
+        batch_size = 13
+        pixel_values = floats_tensor(
+            [
+                batch_size,
+                model.config.vision_config.num_channels,
+                model.config.vision_config.image_size,
+                model.config.vision_config.image_size,
+            ]
+        )
+        input_ids = ids_tensor([batch_size, 4], model.config.text_config.vocab_size)
+        attention_mask = random_attention_mask([batch_size, 4])
+        inputs = {
+            "pixel_values": pixel_values,
+            "input_ids": input_ids,
+            "attention_mask": attention_mask,
+        }
+
+        return model, inputs
+
+    def get_vision_text_model(self, vision_config, text_config):
+        vision_model = FlaxCLIPVisionModel(vision_config)
+        text_model = FlaxBertModel(text_config)
+        return vision_model, text_model
+
+    def prepare_config_and_inputs(self):
```

Comment:
We have a couple of trained BertVIT models from the Flax/JAX sprint - I think we should at least test 1,2 of them here to make sure fine-tuned models work correctly